extension/src/test/e2e/extension.test.ts

```diff
@@ -130,32 +133,74 @@ describe('Experiments Table Webview', function () {
 })
 
 describe('Plots Webview', function () {
+  const webview = new PlotsWebview('plots')
+
   it('should load the plots webview with non-empty plots', async function () {
     this.timeout(60000)
-    const webview = new PlotsWebview('plots')
     const workbench = await browser.getWorkbench()
     await workbench.openCommandPrompt()
     await browser.keys([...'DVC: Show Plots', 'ArrowDown', 'Enter'])
 
     await waitForDvcToFinish()
-
     await webview.focus()
 
-    await browser.waitUntil(
-      async () => {
-        return (await webview.vegaVisualization$$.length) === 5
-      },
-      { timeout: 30000 }
-    )
+    await waitForAllPlotsToRender(webview, 5)
 
     const plots = await webview.vegaVisualization$$
+    for (const plot of plots) {
+      const plotNotEmpty = await webview.plotNotEmpty(plot)
+      expect(plotNotEmpty).toBe(true)
+    }
 
+    await webview.unfocus()
+    await closeAllEditors()
+  })
+
+  it('should create and delete a custom plot', async function () {
+    this.timeout(60000)
+    await createCustomPlot()
+    const workbench = await browser.getWorkbench()
+    await workbench.openCommandPrompt()
+    await workbench.executeCommand('DVC: Show Plots')
+
+    await waitForDvcToFinish()
+    await webview.focus()
+
+    await waitForAllPlotsToRender(webview, 6)
+
+    let plots = await webview.vegaVisualization$$
     for (const plot of plots) {
       const plotNotEmpty = await webview.plotNotEmpty(plot)
       expect(plotNotEmpty).toBe(true)
     }
 
     await webview.unfocus()
+    await closeAllEditors()
+
+    await deleteCustomPlot()
+    await workbench.executeCommand('DVC: Show Plots')
+
+    await waitForDvcToFinish()
+    await webview.focus()
+
+    await waitForAllPlotsToRender(webview, 5)
+
+    plots = await webview.vegaVisualization$$
+    for (const plot of plots) {
+      const plotNotEmpty = await webview.plotNotEmpty(plot)
+      expect(plotNotEmpty).toBe(true)
+    }
+
+    await webview.unfocus()
+    await closeAllEditors()
+  })
+
+  after(async function () {
```

Comment:
[Q] Can this go in the `after` block at the start of the test file? 

Having multiple before/after/beforeEach/afterEach blocks makes things harder to reason about.